File: 1790. Check if One String Swap Can Make Strings Equal/python/main.py
class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        difference_counter = 0
        i, j = -1, -1
        for index, (s1_el, s2_el) in enumerate(zip(s1, s2)):
            if s1_el != s2_el:
                difference_counter += 1
                if i == -1:
                    i = index
                elif j == -1:
                    j = index
                if difference_counter > 2:
                    return False
        return difference_counter == 0 or (difference_counter == 2 and s1[i] == s2[j] and s1[j] == s2[i])

File: 1790. Check if One String Swap Can Make Strings Equal/javascript/main.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {boolean}
 */
var areAlmostEqual = function (s1, s2) {
  let differenceCounter = 0;
//   Create variables to store the index of the difference pair
  let i = -1;
  let j = -1;
  //   Check if
  for (let index = 0; index < s1.length; index++) {
    const s1El = s1[index];
    const s2El = s2[index];
    if (differenceCounter > 2) {
      return false;
    }
    if (s1El !== s2El) {
      differenceCounter++;
      if (i == -1) {
        i = index;
      } else if (j == -1) {
        j = index;
      }
    }
  }

  return (
    differenceCounter === 0 ||
    (differenceCounter === 2 && s1[i] === s2[j] && s1[j] === s2[i])
  );
};
